fv3core/stencils/d2a2c_vect.py

```diff
@@ -379,182 +512,47 @@ def compute(dord4, uc, vc, u, v, ua, va, utc, vtc):
         domain=(idiff, grid.njc + 2, grid.npz),
     )
 
-    domain_edge_x = (1, grid.njc + 2, grid.npz)
-    jslice = slice(grid.js - 1, grid.je + 2)
-    if grid.west_edge and not grid.nested:
-        vol_conserv_cubic_interp_x(utmp, uc, origin=(i1, j1, 0), domain=domain_edge_x)
-        islice = slice(grid.is_ - 2, grid.is_ + 2)
-        for k in range(grid.npz):
-            utc[grid.is_, jslice, k] = edge_interpolate4_x(
-                ua[islice, jslice, k], grid.dxa[islice, jslice]
-            )
-        uc_x_edge1(
-            utc,
-            grid.sin_sg3,
-            grid.sin_sg1,
-            uc,
-            origin=(i1 + 1, j1, 0),
-            domain=domain_edge_x,
-        )
-        vol_conserv_cubic_interp_x_rev(
-            utmp, uc, origin=(i1 + 2, j1, 0), domain=domain_edge_x
-        )
-        contravariant_stencil(
-            uc,
-            v,
-            grid.cosa_u,
-            grid.rsin_u,
-            utc,
-            origin=(i1, j1, 0),
-            domain=domain_edge_x,
-        )
-        contravariant_stencil(
-            uc,
-            v,
-            grid.cosa_u,
-            grid.rsin_u,
-            utc,
-            origin=(i1 + 2, j1, 0),
-            domain=domain_edge_x,
-        )
+    east_west_edges(
+        u,
+        ua,
+        uc,
+        utc,
+        utmp,
+        v,
+        grid.sin_sg1,
+        grid.sin_sg3,
+        grid.cosa_u,
+        grid.rsin_u,
+        grid.dxa,
+        origin=grid.compute_origin(add=(-3, -3, 0)),
+        domain=grid.domain_shape_compute(add=(6, 6, 0)),
```

Comment:
we usually use full_origin and domain_shape_full for these, but you don't need to change it, this is fine